js/batch_ingest.js

```diff
@@ -92,14 +119,53 @@ function openFile(event) {
                 invalidData[invalidCount] = data[j];
                 invalidCount++;
             }
+        }*/
+
+           if(!validateEmail(data, j)) {
+                invalidEmail[invalidEmailCount] = data[j];
+                invalidEmailCount++;
+           }
+           if(!validateID(data, j)) {
+                invalidID[invalidIDCount] = data[j];
+                invalidIDCount++;
+           }
+           if(!validateDate(data, j)) {
+               invalidDate[invalidDateCount] = data[j];
+               invalidDateCount++;
+           }
+           if(!validateCatYear(data, j)) {
+               invalidCataYear[invalidCYCount] = data[j];
+               invalidCYCount++;
+           }
+           if(!validateProg(data, j)) {
+               invalidProg[invalidProgCount] = data[j];
+               invalidProgCount++;
+           }
+           if(!validateDuplicate(data, j)) {
+               dupeID[duplicateIDCount] = data[j];
+               duplicateIDCount++;
+           } else {
+               validatedData[validCount] = data[j];
+               validCount++;
+           }
         }
+
+        /*ADD CURRENT DATE TO VALIDATED DATA*/
         for(let i = 1; i < validatedData.length; i++) {
             validatedData[i][15] = currentDate;
         }
         console.log(data);
         console.log(validatedData);
         console.log(invalidData);
-        window.localStorage.setItem("errors",JSON.stringify(invalidData));
+        /*SEND ERRORS TO ERROR PAGE*/
+        window.localStorage.setItem("emailErrors",JSON.stringify(invalidEmail));
+        window.localStorage.setItem("idErrors",JSON.stringify(invalidID));
+        window.localStorage.setItem("progErrors",JSON.stringify(invalidProg));
+        window.localStorage.setItem("dateErrors",JSON.stringify(invalidDate));
+        window.localStorage.setItem("cataErrors",JSON.stringify(invalidCataYear));
+        window.localStorage.setItem("dupeIdErrors",JSON.stringify(dupeID));
+
+
 
     };
     console.log(input);
```

Comment:
This comment is for the sendToSheet() function. _(I did not look too closely at your code last time since we knew there were still things to add, and since that function has not been changed in this pull request, yet, github won't let me put comments directly in that area.)_
So, since the validation is done by the time you even display the message "Are you sure you want to send...", why not just avoid the issue of later duplicates, and do something like:

```
var submittableRows = data.length - 1; // The data array includes a header row.
if (actualRows != submittableRows) {
  var errors = confirm("Only " + actualRows + " out of " + submittableRows + " are valid. Click 'OK' to see errors, then fix them and try again.");
  if (errors) {
    window.open("errors.html", "_blank");
  }
} else {
  // Insert the code that submits stuff to the sheet.
}
```

Reply:
Implemented